### DISCUSSION TITLE
server side authentication: setAuth(...) makes the user null

### DISCUSSION THREAD

**Discussion by zakaria-chahboun** [supabase/supabase#5881] [Questions]
i have the latest version of: `"@supabase/supabase-js": "^1.31.1",`
i already post the issue in the gotrue-js!
https://github.com/supabase/gotrue-js/issues/243

**Comment by greentfrapp** [comment]
@zakaria-chahboun I had the same issue but found a workaround by using `supabase.auth.signIn({refreshToken})` instead of using `setAuth`, where `refreshToken` can be retrieved from the `Session` object. This will set the user correctly (but also refreshes and extends the session, which might be unintended).

**Reply by zakaria-chahboun** [reply]
That will cause an other issue, I have both authentication in server and client `(for SSR jamstack)`, So it must both sides have the same session!
